pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -142,6 +153,78 @@ func (r *ReconcileMachineHealthCheck) Reconcile(request reconcile.Request) (reco
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileMachineHealthCheck) nodeRequestsFromMachineHealthCheck(o handler.MapObject) []reconcile.Request {
+	glog.V(3).Infof("Watched machineHealthCheck event, finding nodes to reconcile.Request...")
+	mhc := &healthcheckingv1alpha1.MachineHealthCheck{}
+	if err := r.client.Get(
+		context.Background(),
+		client.ObjectKey{
+			Namespace: o.Meta.GetNamespace(),
+			Name:      o.Meta.GetName(),
+		},
+		mhc,
+	); err != nil {
+		glog.Errorf("No-op: Unable to retrieve mhc %s/%s from store: %v", o.Meta.GetNamespace(), o.Meta.GetName(), err)
+		return []reconcile.Request{}
```

Comment:
This wants `[]reconcile.Request{}` as opposed to just `nil`?

Reply:
yeh that's intentional as it's the expectation by controller runtime internals